### PR TITLE
Improve `vms/` tests with `require`

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Factored out of #1453 

## How this works

update tests

## How this was tested

CI

### CODE REVIEW

File: vms/avm/blocks/builder/builder_test.go [hunk 1]
```diff
@@ -487,13 +487,12 @@ func TestBuilderBuildBlock(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			require := require.New(t)
```
[the remark this answers is not in the document]
Comment (dhrubabasu) [reply vms/avm/blocks/builder/builder_test.go:490]: I talked to @StephenButtolph about this off-band briefly last week. Generally, we'll only do this if there's more than one require in the block. Fully agreed on not having to pass t around, I cleanup some other tests to stop passing it around 😀

File: vms/avm/service_test.go [hunk 2]
```diff
@@ -2658,7 +2466,7 @@ func TestServiceGetBlock(t *testing.T) {
 				block.EXPECT().Bytes().Return(blockBytes)
 
 				expected, err := formatting.Encode(formatting.Hex, blockBytes)
-				require.NoError(err)
+				require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (dhrubabasu) [reply vms/avm/service_test.go:2469]: Yeah, this is also in `master` - I didn't change the functionality here. I think that's a good suggestion tho, added it :)

**Review by danlaine — APPROVED** [review]

**Review by gyuho — APPROVED** [review]

File: vms/registry/vm_registry_test.go [hunk 3]
```diff
@@ -56,26 +58,30 @@ func TestReload_Success(t *testing.T) {
 		Return(nil)
 
 	installedVMs, failedVMs, err := resources.vmRegistry.Reload(context.Background())
-	require.ElementsMatch(t, []ids.ID{id3, id4}, installedVMs)
-	require.Empty(t, failedVMs)
-	require.NoError(t, err)
+	require.ElementsMatch([]ids.ID{id3, id4}, installedVMs)
+	require.Empty(failedVMs)
+	require.NoError(err)
```
Comment (StephenButtolph) [inline vms/registry/vm_registry_test.go:63]: Can we move this error check as the first check?

File: vms/registry/vm_registry_test.go [hunk 4]
```diff
@@ -56,26 +58,30 @@ func TestReload_Success(t *testing.T) {
...
 
 	resources.mockVMGetter.EXPECT().Get().Times(1).Return(nil, nil, errTest)
 
 	installedVMs, failedVMs, err := resources.vmRegistry.Reload(context.Background())
-	require.Empty(t, installedVMs)
-	require.Empty(t, failedVMs)
-	require.ErrorIs(t, err, errTest)
+	require.Empty(installedVMs)
+	require.Empty(failedVMs)
+	require.ErrorIs(err, errTest)
```
Comment (StephenButtolph) [inline vms/registry/vm_registry_test.go:78]: Same - can we move the error check higher?

File: vms/registry/vm_registry_test.go [hunk 5]
```diff
@@ -109,15 +115,17 @@ func TestReload_PartialRegisterFailure(t *testing.T) {
...
-	require.Len(t, failedVMs, 1)
-	require.ErrorIs(t, failedVMs[id3], errTest)
-	require.Len(t, installedVMs, 1)
-	require.Equal(t, id4, installedVMs[0])
-	require.NoError(t, err)
+	require.Len(failedVMs, 1)
+	require.ErrorIs(failedVMs[id3], errTest)
+	require.Len(installedVMs, 1)
+	require.Equal(id4, installedVMs[0])
+	require.NoError(err)
```
Comment (StephenButtolph) [inline vms/registry/vm_registry_test.go:122]: Can we move this up?

File: vms/registry/vm_registry_test.go [hunk 6]
```diff
@@ -150,26 +158,30 @@ func TestReloadWithReadLock_Success(t *testing.T) {
...
 
 	resources.mockVMGetter.EXPECT().Get().Times(1).Return(nil, nil, errTest)
 
 	installedVMs, failedVMs, err := resources.vmRegistry.ReloadWithReadLock(context.Background())
-	require.Empty(t, installedVMs)
-	require.Empty(t, failedVMs)
-	require.ErrorIs(t, err, errTest)
+	require.Empty(installedVMs)
+	require.Empty(failedVMs)
+	require.ErrorIs(err, errTest)
```
Comment (StephenButtolph) [inline vms/registry/vm_registry_test.go:178]: Can we move this up?

File: vms/registry/vm_registry_test.go [hunk 7]
```diff
@@ -203,11 +215,11 @@ func TestReloadWithReadLock_PartialRegisterFailure(t *testing.T) {
...
-	require.Len(t, failedVMs, 1)
-	require.ErrorIs(t, failedVMs[id3], errTest)
-	require.Len(t, installedVMs, 1)
-	require.Equal(t, id4, installedVMs[0])
-	require.NoError(t, err)
+	require.Len(failedVMs, 1)
+	require.ErrorIs(failedVMs[id3], errTest)
+	require.Len(installedVMs, 1)
+	require.Equal(id4, installedVMs[0])
+	require.NoError(err)
```
Comment (StephenButtolph) [inline vms/registry/vm_registry_test.go:222]: Can we move this up?

File: vms/proposervm/scheduler/scheduler_test.go [hunk 8]
```diff
@@ -22,9 +24,7 @@ func TestDelayFromNew(t *testing.T) {
 	fromVM <- common.PendingTxs
 
 	<-toEngine
-	if time.Until(startTime) > 0 {
-		t.Fatalf("passed message too soon")
-	}
+	require.Negative(t, time.Until(startTime))
```
[the remark this answers is not in the document]
Comment (dhrubabasu) [reply vms/proposervm/scheduler/scheduler_test.go:27]: Fair, used `LessOrEqual` instead.

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**